src/project.rs

```diff
@@ -931,25 +932,18 @@ pub struct Manifest {
 
 impl Manifest {
     pub fn parse(path: &Path) -> Result<Self> {
-        let contents = fs::read(path)?;
-        let value: toml::Value = toml::from_slice(&contents)?;
-        let package = value
-            .as_table()
-            .and_then(|v| v.get("package"))
-            .and_then(toml::Value::as_table);
-        let crate_name = package
-            .and_then(|v| v.get("name"))
-            .and_then(toml::Value::as_str)
-            .with_context(|| anyhow!("{} (package.name) is malformed", path.display()))?
-            .to_owned();
-        let edition = package
-            .expect("can't be None at this point")
-            .get("edition")
-            .map(|v| match v.as_str() {
-                Some(s) => Ok(s.to_owned()),
-                None => bail!("{} (package.edition) is malformed", path.display()),
-            })
-            .transpose()?;
+        let metatdata = MetadataCommand::new().exec()?;
+        let package = metatdata
+            .packages
+            .iter()
+            .find(|p| p.manifest_path == path)
+            .expect({
+                let path = path.display();
+                &format!("could not find package for {}", path)
+            });
```

Comment:
Rather than panic (via `expect`) can we return actual errors? That will also fix the issue of always allocating the panic message string even when we don't panic here.

Reply:
This was removed in https://github.com/rust-fuzz/cargo-fuzz/pull/349/commits/a4a946e8fd4f945f042691b23f5a04d7f34c98bd